### PR TITLE
TarGzFileExtractToDirectory

### PR DESCRIPTION
**Pull request by piotrpMSFT**
Adds TarGzFileExtractToDirectory
Also cleans up unnecessary override in TarGzFileCreateFromDirectory and out-of-date comments in ZipFileExtractToDirectory

### CODE REVIEW

**Comment by piotrpMSFT** [comment]
@weshaggard thoughts?

File: src/Microsoft.DotNet.Build.Tasks/TarGzFileExtractToDirectory.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                if (OverwriteDestination == true)
+                {
+                    Log.LogMessage(MessageImportance.Low, "'{0}' already exists, trying to delete before unzipping...", DestinationDirectory);
+                    Directory.Delete(DestinationDirectory, recursive: true);
+                }
+                else
+                {
+                    Log.LogWarning("'{0}' already exists. Did you forget to set '{1}' to true?", DestinationDirectory, nameof(OverwriteDestination));
+
+                    retVal = false;
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/TarGzFileExtractToDirectory.cs]: If you are going to make this fail should this be a LogError instead?

**Comment by weshaggard** [comment]
One comment otherwise LGTM

**Comment by piotrpMSFT** [comment]
@weshaggard I was going to drop by to chat about this. The Zip version of this capability LogWarning's so I maintained parity. However, Zip version does not even return false, so though it spits out a warning it will continue along its way to attempt a decompression. I'm glad to make them match one another but was not familiar with Zip's intention.

My proposal is that we make both LogError and both return 'false'. This is in keeping with my expectation as a user.

Returning false is not likely to be strictly necessary but I'd want to introduce the pattern for folks who choose not to LogError in the future.

Thoughts?

**Comment by weshaggard** [comment]
>My proposal is that we make both LogError and both return 'false'. This is in keeping with my expectation as a user.

That seems reasonable to me.

**Comment by Chrisboh** [comment]
No action in any recent amount of time.  Please reopen if needed.

**Pull request closed without merging**